pkg/controller/certmanager/controller.go

```diff
@@ -61,26 +84,54 @@ type CertManagerReconciler struct {
 //+kubebuilder:rbac:groups="acme.cert-manager.io",resources=challenges;challenges/finalizers;challenges/status;orders;orders/finalizers;orders/status,verbs=get;list;watch;create;update;patch;delete;deletecollection
 //+kubebuilder:rbac:groups="route.openshift.io",resources=routes;routes/custom-host,verbs=get;list;watch;create;update;patch;delete
 
-// Reconcile is part of the main kubernetes reconciliation loop which aims to
-// move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the CertManager object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
+// Reconcile is run each time the CertManager object has a create/update/delete event.
+// All objects other than certmanager.operator.openshift.io/cluster lead to no-op syncs.
 func (r *CertManagerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```

Comment:
nit: We could add couple of UTs here I think.

Reply:
Added few test cases for the basic functionality, now in `tp_status_test.go`.